libs/patchwheel/patch.go

```diff
@@ -0,0 +1,280 @@
+package patchwheel
+
+import (
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+const (
+	versionKey = "Version:"
+	nameKey    = "Name:"
+)
+
+// readMetadataAndRecord scans the zip file for files matching the patterns
+// "*.dist-info/METADATA" and "*.dist-info/RECORD". If multiple are found, it picks
+// the first one encountered.
+func readMetadataAndRecord(r *zip.ReadCloser) (metadataFile, recordFile *zip.File, oldDistInfoPrefix string) {
+	for _, f := range r.File {
+		if metadataFile == nil {
+			matched, _ := path.Match("*.dist-info/METADATA", f.Name)
```

Comment:
Are the slashes the same on Windows in this case?

Reply:
yes, slashes are forward (also, we run the test on windows).